framework/tlvf/src/include/tlvf/BaseClass.h

```diff
@@ -38,6 +37,19 @@ class BaseClass {
     const bool m_parse__;
     const bool m_swap__;
     bool m_init_succeeded = false;
+
+private:
+    /**
+     * @brief trim buffer by tailroom size
+     *
+     * This is intended to be used __only__ by friend class TlvList on finalize
+     * When resizing buffers.
+     *
+     * @param tailroom size to trim
+     * @return true on success
+     * @return false on failure
+     */
+    friend void trim(BaseClass &obj, size_t tailroom) { obj.m_buff_ptr__ -= tailroom; };
```

Comment:
I think the trim is not appropriate here. I think it would make more sense to make a special case in tlvf for when an unknown-length list is at the end of the class (or maybe even make a special _payload type for this). In that specific case, we could add two functions: a prepare that reserves the rest of the buffer and returns the remaining length, and a commit with the final length as argument that updates the end pointer and the length field. Both functions are pretty much the same as allocate (the trim is like allocate with a negative value) in that they update both length and end pointer.

Reply:
Sound OK, but the commit API doesn't need an argument, it knows the buffer length and we know the current buffer pointer so we can calculate that ourselves